autogluon/task/text_classification/dataset.py

```diff
@@ -233,37 +240,53 @@ def get_dataset(self, segment='train'):
         elif segment == 'dev':
             return SampledDataset(self.dataset, self.dev_sampler)
         else:
-            raise NotImplementedError
+            raise ValueError('Please specify either train or dev dataset.')
 
     def _read(self, **kwargs):
-        path = kwargs.get('filepath', None)
-        kwargs['filepath_or_buffer'] = path
-        kwargs.pop("filepath")
         low_memory = kwargs.get('low_memory', None)
         if low_memory is None:
             kwargs['low_memory'] = False
-        dataset_df = pd.read_csv(**kwargs)
+        split_ratio = kwargs.get('split_ratio', None) if 'split_ratio' in kwargs else 0.8
+        if split_ratio is None:
+            split_ratio = 0.8
+        if 'split_ratio' in kwargs:
+            kwargs.pop('split_ratio')
+        path = kwargs.get('filepath', None)
+        kwargs['filepath_or_buffer'] = path
+        dataset_df = kwargs.get('df', None)
+        if path is not None:
```

Comment:
path should not silently override df, either raise exception if both are passed or prefer DF, since it is already loaded.